keps/sig-api-machinery/4222-cbor-serializer/README.md

```diff
@@ -613,7 +620,143 @@ sequence that it will not successfully decode.
+preparing to send, or handle a received object containing `RawExtension`, callers can assume that
+the Raw bytes are in the same encoding as the negotiated request or response encoding.
+
+##### Stored External Types
+
+Storing the verbatim Raw bytes of a `RawExtension` received from a client introduces additional
+considerations on top of the transient (transmit-only) case. The encoding of the Raw bytes is
+determined by encoding of the request that wrote the value of the RawExtension, which may or may not
+be the same as the object's storage encoding.
+
+##### Types as Canonical Definition of Custom Resources
+
+Throughout the ecosystem, it is common practice to maintain Go structs as the canonical definition
+for API extensions. In many cases, `controller-gen` is used to mechanically translate such types
+from Go sources to CustomResourceDefinition manifests. Similarly, `client-gen` can produce typed Go
+clients that use the canonical Go types directly. These Go struct types can and sometimes do include
+fields of type `RawExtension`
+([example](https://github.com/openshift/api/blob/944467d2cc3b03225ccc24c4e88b876396202d5a/operator/v1/types.go#L91)).
+
+#### Scenarios
+
+The following table enumerates API request and response flows that can involve `RawExtension`.
+
+The *Client* and *Server* columns indicate the types the named component uses to processes API
+objects. If "dynamic", it uses Unstructured (e.g. a custom resource handler or a dynamic client). If
+"static", it uses API-specific Go types that may include `RawExtension` (e.g. clients generated by
+`client-gen`, kube-apiserver built-in types, aggregated apiservers). The table omits cases where
+both the client and the server are dynamic (e.g. a dynamic client and a custom resource handler),
+since neither side should be dealing with `RawExtension` values. The edge case where a client
+program makes a `RawExtension` a child of an Unstructured value's `map[string]interface{}` can be
+considered a static client case for the purposes of this evaluation.
+
+| N  | Client  | Server  | Direction | Encoding |
+|----|---------|---------|-----------|----------|
+| 1  | static  | static  | request   | json     |
+| 2  | static  | static  | request   | cbor     |
+| 3  | static  | static  | response  | json     |
+| 4  | static  | static  | response  | cbor     |
+| 5  | dynamic | static  | request   | json     |
+| 6  | dynamic | static  | request   | cbor     |
+| 7  | dynamic | static  | response  | json     |
+| 8  | dynamic | static  | response  | cbor     |
+| 9  | static  | dynamic | request   | json     |
+| 10 | static  | dynamic | request   | cbor     |
+| 11 | static  | dynamic | response  | json     |
+| 12 | static  | dynamic | response  | cbor     |
+| 13 | static  | static  | request   | protobuf |
+| 14 | static  | static  | response  | protobuf |
+
+In cases 5-6 and 11-12, the marshalling side acts on an Unstructured object and is not aware that
```

Comment:
why is server in case 5-6 operating on an Unstructured?  I thought that static meant API-specific go types.